llvm/include/llvm/ProfileData/SampleProfReader.h

```diff
@@ -604,48 +604,52 @@ class SampleProfileReaderBinary : public SampleProfileReader {
   /// EC is set.
   ///
   /// \returns the read value.
-  template <typename T> ErrorOr<T> readNumber();
+  template <typename T> sampleprof_error readNumber(T &Result);
```

Comment:
I only see changes to the function declaration but not int he function implementation. Am I missing anything or it'll be included in a separate PR?

I also wonder how much end-to-end performance improvement did you see in your experiments? The `ErrorOr` usage is quite handy and prevailing in the code base.

Reply:
The implementation is in the cpp file. This is how the original code is like, since it is a protected function, as long as some other functions in the cpp file instantiate the template, the code will compile fine

Reply:
I will do some more benchmarking to see the impact